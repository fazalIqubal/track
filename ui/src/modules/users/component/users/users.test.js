import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import  Users  from './users'
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson} from 'enzyme-to-json';
import Search from 'antd/lib/input/Search';

Enzyme.configure({ adapter: new Adapter() })
describe('when page is rendering', () => {
    let userDataComponent
    const mockDispatch = jest.fn();
    beforeEach(() => {
        userDataComponent = shallow(<Users.WrappedComponent usersData ={{ data: [] }} dispatch={mockDispatch}/>)
    })
    afterEach(() => {
        userDataComponent = null
    })
    
    it('should render the form', ()=> {
        expect(shallowToJson(userDataComponent)).toMatchSnapshot();
    })
    it('should have a heading text', ()=> {
        expect(userDataComponent.find('h3').text()).toBe('Users');
    })
    it('should have a input box ', ()=> {
        expect(userDataComponent.find('.search-box').exists()).toEqual(true);
    })
    it('should have a input box ', ()=> {
        expect(userDataComponent.find('.search-box').props().placeholder).toBe('Search ');
    })
    it('should have a button', ()=> {
        expect(userDataComponent.find('Button').exists()).toEqual(true);
    })
    it('should have text in button', ()=> {
        expect(userDataComponent.find('Button').text()).toEqual('Create User');
    })
    it('should render the usertable', ()=> {
        expect(userDataComponent.find('.users-table').exists()).toEqual(true);
    })
    it('should render the usertable', ()=> {
        expect(userDataComponent.find('.user-form-container').exists()).toEqual(true);
    })
    
})
