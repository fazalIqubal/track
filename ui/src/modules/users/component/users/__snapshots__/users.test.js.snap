// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when page is rendering should render the form 1`] = `
<div
  className="user-table-container"
>
  <h3>
    Users
  </h3>
  <hr />
  <div
    className="table-wrapper"
  >
    <Row>
      <Col
        span={22}
      >
        <Input
          className="search-box"
          enterButton={true}
          onChange={[Function]}
          placeholder="Search "
          prefix={<ForwardRef(SearchOutlined) />}
          type="text"
        />
      </Col>
      <Col
        span={2}
      >
        <Button
          block={false}
          className="create-user-btn"
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
          type="primary"
        >
          Create User
        </Button>
      </Col>
    </Row>
    <TableComponent
      className="users-table"
      columns={
        Array [
          Object {
            "className": "columnTitle",
            "dataIndex": "first_name",
            "ellipsis": true,
            "isFilters": true,
            "key": "first_name",
            "title": "First Name",
            "width": 200,
          },
          Object {
            "className": "columnTitle",
            "dataIndex": "last_name",
            "ellipsis": true,
            "isFilters": true,
            "key": "last_name",
            "title": "Last Name",
          },
          Object {
            "className": "columnTitle",
            "dataIndex": "address",
            "ellipsis": true,
            "isFilters": true,
            "key": "address",
            "title": "Address",
            "width": 250,
          },
          Object {
            "className": "columnTitle",
            "dataIndex": "username",
            "ellipsis": true,
            "isFilters": true,
            "key": "username",
            "title": "Username",
          },
          Object {
            "className": "columnTitle",
            "dataIndex": "mobile",
            "isFilters": true,
            "key": "mobile",
            "title": "Mobile",
          },
          Object {
            "className": "columnTitle",
            "dataIndex": "email",
            "ellipsis": true,
            "isFilters": true,
            "key": "email",
            "title": "Email",
            "width": 250,
          },
          Object {
            "key": "action",
            "render": [Function],
            "title": "",
          },
        ]
      }
      dataSource={Array []}
    />
  </div>
  <FormSlideout
    className="user-form-container"
    form="users-form"
    formElement={
      <UNDEFINED
        isEdit={false}
      />
    }
    onClose={[Function]}
    primaryBtnEvent={[Function]}
    primary_button_name="Save"
    title="Create New User"
    visible={false}
  />
  <FormSlideout
    className="user-form-container"
    form="users-form"
    formElement={
      <UNDEFINED
        isEdit={false}
      />
    }
    onClose={[Function]}
    primaryBtnEvent={[Function]}
    primary_button_name="Save"
    title="Edit User"
    visible={false}
  />
</div>
`;
